castle/cms/static/scripts/slide.js

```diff
@@ -0,0 +1,88 @@
+function getFormVariables(event) {
+    const form = document.getElementById('edit_tile');
+    if (!form) {
+        return;
+    }
+    const typeAndTextFieldset = document.getElementById('fieldset-type-and-text');
+    const mediaFieldset = document.getElementById('fieldset-media-settings');
+    const formVariables = {
+        form,
+        typeAndTextFieldset,
+        mediaFieldset,
+        formFields: [
+            ...Array.from(typeAndTextFieldset.children).filter(hasId),
+            ...Array.from(mediaFieldset.children).filter(hasId)
+        ],
+        navTabs: form.querySelector('nav')
+    };
+    if (event) {
+        formVariables.slideType = event.target.value;
+    } else {
+        const slideTypeSelect = form.querySelector("select[id$='widgets-display_type'");
+        formVariables.slideType = slideTypeSelect.options[slideTypeSelect.selectedIndex].value;
+    }
+    return formVariables;
+}
+function shouldElementBeHidden(id, idEndingsToHide) {
+    console.log(`id: ${id}, endings to hide: ${idEndingsToHide}`)
+    console.log(idEndingsToHide.some((ending) => id.endsWith(ending)));
+    return idEndingsToHide.some(ending => id.endsWith(ending));
+}
+function hideFields(form) {
+    const { slideType, formFields } = form;
+    const idEndingsToHide = getIdEndingsToHide(slideType);
+    console.log(`slide type: ${slideType}, endingsToHide: ${idEndingsToHide}`);
+    formFields.forEach((formField) => {
+      const id = formField.id;
+      if (shouldElementBeHidden(id, idEndingsToHide)) {
+        formField.hidden = true;
+      } else {
+        formField.hidden = false;
+      }
+    });
+}
+function getIdEndingsToHide(slideType) {
+    switch (slideType) {
+        case 'background-image':
+        case 'left-image-right-text':
+            return ['widgets-video', 'related_items'];
+        case 'background-video':
+        case 'left-video-right-text':
+            return ['widgets-image', 'related_items'];
+        case 'resource-slide':
+            return ['widgets-image', 'widgets-video'];
+        default:
+            return [];
+    }
+}
+function onSlideTypeChange(event) {
+    console.log(event);
+    const form = getFormVariables(event);
+    console.log(form);
+    hideFields(form);
+}
+function hasId(element) {
+    return !!element.id;
+}
+function modifyEditTile(mutations) {
+    const form = getFormVariables();
+    if (form) {
+        console.log(form.slideType);
+        hideFields(form);
+        // updateFieldsets(form);
+    }
+}
+function observe() {
+    const observer = new MutationObserver(modifyEditTile);
+    const body = document.querySelector('body');
+    const observerOptions = { childList: true, attributes: false, subtree: true };
+    observer.observe(body, observerOptions);
+}
+
+
+
+// castle attempts to execute this script multiple times, so block the rest
+if (!window.scriptCalled) {
```

Comment:
Since window is global we might call this slideUpdateScriptCalled or something just to be sure there's no conflicts anywhere in the future.

Reply:
sounds like a good idea